src/base.cc

```diff
@@ -1,6 +1,61 @@
 #include "astshim/base.h"
 
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
 namespace ast {
+namespace {
+
+    static std::ostringstream errorMsgStream;
+
+    /*
+    Write an error message to `errorMsgStream`
+
+    Intended to be registered as an error handler to AST by calling `astSetPutErr(reportError)`.
+    */
+    void reportError(int errNum, const char * errMsg) {
+        errorMsgStream << errMsg;
+    }
+
+    /*
+    Instantiate this class to register `reportError` as an AST error handler.
+    */
+    class ErrorHandler {
+    public:
+        ErrorHandler() {
+            astSetPutErr(reportError);
+        }
+
```

Comment:
Is `astshim` bound by [RFC-209](https://jira.lsstcorp.org/browse/RFC-209)? I'm not clear on its relationship to the Stack.